nodes/nomos-node/node/src/config/mod.rs

```diff
@@ -98,25 +97,32 @@ pub struct HttpArgs {
 #[derive(Parser, Debug, Clone)]
 pub struct CryptarchiaArgs {
     #[clap(
-        long = "consensus-note-sk",
-        env = "CONSENSUS_NOTE_SK",
-        requires("note_value")
+        long = "consensus-utxo-sk",
+        env = "CONSENSUS_UTXO_SK",
+        requires = "value"
     )]
-    note_secret_key: Option<String>,
+    pub secret_key: Option<String>,
 
     #[clap(
-        long = "consensus-note-value",
-        env = "CONSENSUS_NOTE_VALUE",
-        requires("note_secret_key")
+        long = "consensus-utxo-value",
+        env = "CONSENSUS_UTXO_VALUE",
+        requires = "secret_key"
     )]
-    note_value: Option<u32>,
+    value: Option<u64>,
 
     #[clap(
-        long = "consensus-note-nonce",
-        env = "CONSENSUS_NOTE_NONCE",
-        requires("note_value")
+        long = "consensus-utxo-txhash",
+        env = "CONSENSUS_UTXO_TXHASH",
+        requires = "value"
     )]
-    note_nonce: Option<String>,
+    tx_hash: Option<String>,
+
+    #[clap(
+        long = "consensus-utxo-output-index",
+        env = "CONSENSUS_UTXO_OUTPUT_INDEX",
+        requires = "value"
+    )]
+    output_index: Option<usize>,
```

Comment:
Nitpick: Maybe we can encapsulate this in its own utxo related config? Then re-embedded it here

Reply:
I wanted to, but you can't flatten a struct and make it optional with clap, so this is a bit of machinery to achieve something like:

```
struct Utxo {
  value: ...
  sk: ...
  ....
}


struct CryptarchiaArgs {
    utxo: Option<Utxo> // clap does not understand this is optional and still requires all fields
}
```
if that's what you meant